perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp

```diff
@@ -177,7 +179,11 @@ void RoiClusterFusionNode::fuseOnSingleImage(
       if (use_iou_y_) {
         iou_y = calcIoUY(cluster_map.second, feature_obj.feature.roi);
       }
-      if (max_iou < iou + iou_x + iou_y) {
+      if (
+        max_iou < iou + iou_x + iou_y &&
+        ((!only_allow_inside_cluster_) ||
+         (only_allow_inside_cluster_ &&
+          is_inside(feature_obj.feature.roi, cluster_map.second, roi_scale_factor_)))) {
```

Comment:
How about this?
```suggestion
      const bool passed_inside_cluster_gate =  only_allow_inside_cluster_ &&
          is_inside(feature_obj.feature.roi, cluster_map.second, roi_scale_factor_)
      if (max_iou < iou + iou_x + iou_y && passed_inside_cluster_gate) {
```

Reply:
If those, doesn't `passed_inside_cluster` always return `false` when `only_allow_inside_cluster=false`?

Reply:
Exactly
How about this?
```suggestion
      const bool passed_inside_cluster_gate = only_allow_inside_cluster_? is_inside(feature_obj.feature.roi, cluster_map.second, roi_scale_factor_) : true;
      if (max_iou < iou + iou_x + iou_y && passed_inside_cluster_gate) {
```

Reply:
Looks good!